### PR TITLE
Fix behaviour of OrderedQ 

### PR DESCRIPTION
**Pull request by adamantinum**
Fix issue https://github.com/Mathics3/mathics-core/issues/201 changing argument of the function to list

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/207"></a>

### CODE REVIEW

File: mathics/builtin/structure.py [hunk 1]
```diff
@@ -291,24 +291,27 @@ def apply(self, p1, p2, evaluation):
...
-    def apply(self, e1, e2, evaluation):
-        "OrderedQ[e1_, e2_]"
+    def apply(self, expr, evaluation):
+        "OrderedQ[expr_]"
 
-        if e1 <= e2:
-            return SymbolTrue
-        else:
-            return SymbolFalse
+        for i in range(1, len(expr.leaves)):
```
Comment (TiagoCavalcante) [inline mathics/builtin/structure.py:308]: This idiom is not recommended in Python, could you replace it with [enumerate](https://realpython.com/python-enumerate/)?

**Review by TiagoCavalcante — CHANGES_REQUESTED** [review]
LGTM, but this should be backward compatible. Could you add an "if there is only 1 leave and that is a list then [new behavior] else [old behavior]"?

**Comment by adamantinum** [comment]
> LGTM, but this should be backward compatible. Could you add an "if there is only 1 leave and that is a list then [new behavior] else [old behavior]"?

Rewritten with enumerate.
Regarding the other point I have a doubt: the old behaviour is not a correct statement in mathematica (i.e.: `Nonatomic expression expected at position 1 in OrderedQ[a, b].`), is it ok?

**Comment by TiagoCavalcante** [comment]
> > LGTM, but this should be backward compatible. Could you add an "if there is only 1 leave and that is a list then [new behavior] else [old behavior]"?
> 
> Rewritten with enumerate. Regarding the other point I have a doubt: the old behaviour is not a correct statement in mathematica (i.e.: `Nonatomic expression expected at position 1 in OrderedQ[a, b].`), is it ok?

Yes, I'll merge this.

**Review by TiagoCavalcante — APPROVED** [review]

**Comment by adamantinum** [comment]
> > > LGTM, but this should be backward compatible. Could you add an "if there is only 1 leave and that is a list then [new behavior] else [old behavior]"?
> > 
> > 
> > Rewritten with enumerate. Regarding the other point I have a doubt: the old behaviour is not a correct statement in mathematica (i.e.: `Nonatomic expression expected at position 1 in OrderedQ[a, b].`), is it ok?
> 
> Yes, I'll merge this.

Thus no need to maintain the old behaviour?

**Comment by TiagoCavalcante** [comment]
@adamantinum no, that is very weird and wrong, that receives 2 parameters, and as its name isn't Ordered**Pair** that is just wrong, you should replace that with a LessEqual sign in those cases.

**Comment by adamantinum** [comment]
> @adamantinum no, that is very weird and wrong, that receives 2 parameters, and as its name isn't Ordered**Pair** that is just wrong, you should replace that with a LessEqual sign in those cases.
I agree!

**Pull request merged**